test/stateful_dataloader/test_dataloader.py

```diff
@@ -1947,6 +1960,116 @@ def test_sampler_reproducibility(self):
+
+        dataset = self.dataset
+        sampler = StatefulDistributedSampler(dataset, num_replicas=10, rank=0)
+        sampler.yielded = 5
+        state_dict = sampler.state_dict()
+        self.assertEqual(state_dict["yielded"], 5)
+
+    def test_load_state_dict(self):
+        from torchdata.stateful_dataloader.sampler import StatefulDistributedSampler
+
+        dataset = self.dataset
+        sampler = StatefulDistributedSampler(dataset, num_replicas=10, rank=0)
+        sampler.load_state_dict({"yielded": 3})
+        self.assertEqual(sampler.next_yielded, 3)
+        with self.assertRaises(ValueError):
+            sampler.load_state_dict({"yielded": -1})
+
+    def test_next_yielded(self):
+        from torchdata.stateful_dataloader.sampler import StatefulDistributedSampler
+
+        sampler = StatefulDistributedSampler(self.dataset, num_replicas=2, rank=0, shuffle=True, seed=42)
+        iterator = iter(sampler)
+        next(iterator)  # advance the iterator
+        self.assertEqual(sampler.yielded, 1)
+        self.assertIsNone(sampler.next_yielded)
+        sampler.load_state_dict({StatefulDistributedSampler._YIELDED: 5})
+        self.assertEqual(sampler.next_yielded, 5)
+        next(iterator)  # advance the iterator again
+        self.assertEqual(sampler.yielded, 6)
+
+    def test_drop_last_effect(self):
+        from torchdata.stateful_dataloader.sampler import StatefulDistributedSampler
+
+        sampler_with_drop = StatefulDistributedSampler(self.dataset, num_replicas=3, rank=0, drop_last=True)
+        sampler_without_drop = StatefulDistributedSampler(self.dataset, num_replicas=3, rank=0, drop_last=False)
+        indices_with_drop = list(iter(sampler_with_drop))
+        indices_without_drop = list(iter(sampler_without_drop))
+        self.assertTrue(
+            len(indices_with_drop) <= len(indices_without_drop), "Drop last should result in fewer or equal indices"
+        )
+
+    def test_data_order_with_shuffle(self):
+        from torchdata.stateful_dataloader.sampler import StatefulDistributedSampler
+
+        sampler = StatefulDistributedSampler(self.mockdataset, num_replicas=1, rank=0, shuffle=True, seed=42)
+        indices = list(iter(sampler))
+        data_sampled = [self.mockdataset[i] for i in indices]
+        self.assertNotEqual(data_sampled, list(range(100)), "Data should be shuffled")
+
+    def test_data_order_without_shuffle(self):
+        from torchdata.stateful_dataloader.sampler import StatefulDistributedSampler
+
+        sampler = StatefulDistributedSampler(self.mockdataset, num_replicas=1, rank=0, shuffle=False)
+        indices = list(iter(sampler))
+        data_sampled = [self.mockdataset[i] for i in indices]
+        self.assertEqual(data_sampled, list(range(100)), "Data should be in sequential order when shuffle is False")
+
+    def test_data_distribution_across_replicas(self):
+        from torchdata.stateful_dataloader.sampler import StatefulDistributedSampler
```

Comment:
Let's move this import to top level for all cases

Reply:
Done.